Patch RichLabelAnnotator, fix in-place functionality of other annotators

LinasKo opened the pull request:
# Description

This is a 3-in-1:
1. Addressed faults with RichLabelAnnotator left after #1116 
  * Primarily: it would not convert back to `np.ndarray` after annotation
2. Realized many annotators do nothing when used in-place.
  * Description on [Loom](https://www.loom.com/share/91308392c7de44ce9ad115778bc99c42?sid=8279b66f-4786-42de-9e77-58cd7381f6be)
  * Fixed in [Colab](https://colab.research.google.com/drive/1NvVs28MnSb-_W-zkFa6-FXiY-0G4nx7k#scrollTo=dVZIKNV1uXVl)
3. Color Annotator would fill boxes with solid color if used in-place, with numpy.

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Ran every annotator except for TraceAnnotator.
Ran tests.
Colab: https://colab.research.google.com/drive/1NvVs28MnSb-_W-zkFa6-FXiY-0G4nx7k#scrollTo=dVZIKNV1uXVl

## Any specific deployment considerations

## Docs

-   [ ] Docs updated? What were the changes:

SkalskiP commented:
@LinasKo I see a lot of `copy to` and `copy` operations. Have you measured how those impact performance?

LinasKo commented:
No. I'll do that, will let you know the results.

Realistically, it should be equivalent to frame.copy().

LinasKo commented:
1. Renamed to `ensure_cv2_image_...` and `ensure_pil_image_...`. I've seen this before, and we're not necessarily doing a conversion.
2. Removed a few copy operations. I finally understand why `cv2` has the destination parameter in methods like `addWeighted`, despite returning an image :)

The Colab now has tests in the end, comparing the performance on a 10,000x10,000 size empty image.
Results should show performance dips if any copying is expensive.

With the latest version, performance is equivalent or slightly better than in base supervision.

LinasKo commented:
Hey @onuralpszr ,

Given you've worked on it before, would you mind reviewing these changes too? There's a colab that shows how it works, only requiring that you upload the font file.

onuralpszr commented:
@LinasKo start reviewing
